internal/v3/storage/valkey.go

```diff
@@ -308,6 +308,86 @@ func (s *ValkeyStorage) handlePubSub() {
 	}
 }
 
+// AddConnection stores connection info in Valkey with TTL
+// Key pattern: conn:{clientID}:{ip}:{origin}
+func (s *ValkeyStorage) AddConnection(ctx context.Context, conn ConnectionInfo, ttl time.Duration) error {
+	log := log.WithField("prefix", "ValkeyStorage.AddConnection")
+
+	key := fmt.Sprintf("conn:%s:%s:%s", conn.ClientID, conn.IP, conn.Origin)
+
+	data := map[string]interface{}{
+		"ip":         conn.IP,
+		"origin":     conn.Origin,
+		"user_agent": conn.UserAgent,
+		"created_at": time.Now().Unix(),
+	}
+
+	pipe := s.client.Pipeline()
+	pipe.HSet(ctx, key, data)
+	pipe.Expire(ctx, key, ttl)
+
+	// Also add to clientID index for efficient lookup
+	indexKey := fmt.Sprintf("conn:idx:%s", conn.ClientID)
```

Comment:
We have conn:%s above, we should have same `nesting` for keys.
For example "conn:full:%s:%s:%s" and "conn:idx:%s"
That way we can range over keys and only get wanted ones